bucoffea/plot/util.py

```diff
@@ -266,6 +262,21 @@ def merge_datasets(histogram):
         for name, regex in yearly.items():
             mapping[name.format(year=year)] = [x for x in all_datasets if re.match(regex.format(year=year), x)]
 
+    # Mapping good to go to be used in "merge_datasets" function
+    return mapping
+
+def merge_datasets(histogram):
+    """Merge datasets that belong same physics process
+
+    :param histogram: The histogram to modify
+    :type histogram: Coffea histogram
+    :return: Modified histogram
+    :rtype: Coffea histogram
+    """
+    all_datasets = list(map(str, histogram.identifiers('dataset')))
+    # Create the mapping for the datasets in this histogram
+    mapping = create_dataset_mapping(all_datasets)
+    
     # Remove empty lists
     tmp = {}
```

Comment:
Why did you choose to keep this cleanup code in merge_datasets? wouldnt it make sense to also move it to create_dataset_mapping?

Reply:
Ah you're right, it would make more sense to move it to create_dataset_mapping, I'll make that change soon